CMakeLists.txt

```diff
@@ -113,12 +112,19 @@ if (ENABLE_HTTP OR ENABLE_DATA_TOOLS)
   endif()
 endif()
 
+# prefer CONFIG mode over MODULE mode, which versions configuration on the package, not CMake
+# NOTE: this is only supported for cmake >= 3.15, but shouldn't be a problem in real life
+set(CMAKE_FIND_PACKAGE_PREFER_CONFIG ON)
```

Comment:
I put this now way up top so all `find_package` calls will prefer CONFIG mode if available (i.e. the appropriate cmake module was part of the dependency's installation)